### DISCUSSION TITLE
private_openssl_ec_diffie_hellman_t

### DISCUSSION THREAD

**Discussion by keskinadem** [strongswan/strongswan#1711] [Q&A]
Hello I was using 5.6.0 version of strongswan. Now, I want to use version 6.0.0. Why there is a  change in **struct private_openssl_ec_diffie_hellman_t**

**Comment by tobiasbrunner** [comment]
Not sure what you mean, but if you are interested in changes to a file, just look [at its history](https://github.com/strongswan/strongswan/commits/six-beta/src/libstrongswan/plugins/openssl/openssl_ec_diffie_hellman.c) or [blame it](https://github.com/strongswan/strongswan/blame/six-beta/src/libstrongswan/plugins/openssl/openssl_ec_diffie_hellman.c).

**Reply by keskinadem** [reply]
For example parameter of EC_KEY changed as EVP_KEY. As I understand it is about openssl and some deprecations. If it is, Is there any minimum required openssl version for latest strongswan version for compatibility?

**Reply by tobiasbrunner** [reply]
Since f26639de2caecebe7afeefaedd270ba4504d2f1a, 1.0.2 is the minimum version that's technically supported by the _openssl_ plugin. However, as the OpenSSL project doesn't maintain versions older than 1.1.1 anymore, and that version also only until September 2023 (see [here](https://www.openssl.org/source/)), I wouldn't recommend using anything other than 3.x (3.0 is an LTS version that's maintained until September 2026).
